### PR TITLE
Added XLMRobertaModel for Reranking

### PR DESCRIPTION
**Pull request by akshayballal95**
Some of the good reranker models are based on this architecture and I think its a good addition to the current repository of models. 

The architecture is not very different than Bert but the position embedding technique is a bit different.

### CODE REVIEW

File: candle-examples/examples/xlm-roberta/main.rs [hunk 1]
```diff
@@ -0,0 +1,283 @@
...
+    model_id: Option<String>,
+
+    #[arg(long, default_value = "main")]
+    revision: String,
+
+    #[arg(long, default_value = "bge-reranker-base")]
+    model: Model,
+
+    #[arg(long, default_value = "reranker")]
+    task: String,
```
Comment (LaurentMazare) [inline candle-examples/examples/xlm-roberta/main.rs:44]: Let's have an enum rather than a String for the task as there seems to only be two possibilities.

File: candle-transformers/src/models/xlm_roberta.rs [hunk 2]
```diff
@@ -0,0 +1,545 @@
...
+        self.classifier.forward(&hidden_states)
+    }
+}
+
+fn prepare_4d_attention_mask(
+    mask: &Tensor,
+    dtype: DType,
+    tgt_len: Option<usize>,
+) -> Result<Tensor> {
+    let bsz = mask.dims()[0];
```
Comment (LaurentMazare) [inline candle-transformers/src/models/xlm_roberta.rs:524]: Use `let bsz = mask.dim(0)?` instead, this will make a proper error rather than a panic if the shape is not correct. Same for the line below.

**Comment by LaurentMazare** [comment]
Thanks!
